app/domain/authentication/authn_jwt/configuration_interface.rb

```diff
@@ -2,6 +2,7 @@ module Authentication
   module AuthnJwt
     # Interface containing JWT configuration functions to be implemented in JWTConfiguration class for a vendor
     class ConfigurationInterface
+      def fetch_token(authentication_parameters); end
```

Comment:
WDYT to put this function `fetch_token` and the general function in the generic code `create_authentication_parameters` under the same umbrella ?

I think in the future each authenticator will have its own logic of create and validate the authenticate parameters, and it will also clean the generic code.

So my idea to change the function in the interface to `validate_and_create_authentication_parameters` 
it responsibility will be to validate the input (in this case the jwt token) and to return a new validated object `authentication_parameters` to be used in all other functions

Reply:
I'm agree. I'll rename the interface method but code rearrange let's do in one of the next PRs